### DISCUSSION TITLE
Are autogenerated commits messages on the horizion for Copilot?

### DISCUSSION THREAD

**Discussion by dimitur2204** [community/community#58035] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

After seeing all the [Copilot CLI ](https://githubnext.com/projects/copilot-cli)and [Copilot4Prs](https://githubnext.com/projects/copilot-for-pull-requests) I am indeed quite excited about the future of Copoilot.

However I can't stop thinking about the missed opportunity on generated commit messages based on the staged files. I **did not see that** mentioned anywhere on both the projects mentioned above! Something that might fit quite well is helping [Copilot4Prs](https://githubnext.com/projects/copilot-for-pull-requests) generate better PRs based on these generated commit messages.

I personally spend a lot of time on thinking of meaningful commit messages, it's one of the most annoying things in my workflow. I have found this open source project doing just that https://github.com/di-sukharev/opencommit. 
It looks incredible already and I am certain with the powerful tools the Github team has access to, they can so something better and more seamlessly integrated with IDEs and terminals!

**Comment by Sandro642** [comment]
Hello, @dimitur2204 

I recently came across the Copilot CLI and Copilot4Prs projects, and I must say I'm excited about their potential. However, I couldn't help but notice the absence of generated commit messages based on staged files. I believe this is a missed opportunity.

Generating meaningful commit messages is often a time-consuming and tedious task in my workflow. It would be great if Copilot4Prs could assist in generating better pull requests (PRs) based on these generated commit messages.

I came across an open-source project called OpenCommit (https://github.com/di-sukharev/opencommit) that addresses this issue. It already looks impressive, but with the powerful tools available to the GitHub team, I'm certain they can create something even better and seamlessly integrate it with IDEs and terminals.

I wanted to suggest exploring the possibility of enhancing Copilot4Prs with improved commit message generation capabilities. It would greatly benefit developers like me who value clear and meaningful commit messages.

Thank you for considering my suggestion. I look forward to seeing the continued development of Copilot.

Best regards,
Sandro642

Please answered my responses if is good ;)
